infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutAlignmentDao.kt

```diff
@@ -329,6 +363,8 @@ fun getSegmentPoints(
     heightColumn: String,
     cantColumn: String,
 ): List<LayoutPoint> {
+    val rawGeometry = rs.getString(geometryColumn)
+    if (rawGeometry == null) return emptyList()
```

Comment:
Korjaus vanhaan koodiin. Vaikka segmenttien `geometry`-columni onkin non-nullable, voi postgisin `st_astext`-funktio palauttaa nullin tilanteissa joissa geometriaa ei ole (esim. alignment jolla haetaan on vasta luotu sijaintiraide). Tätä funktiota ei käytetä kauheasti muualta, joten ei sen takia liene tullut aiemmin vastaan.

Reply:
Ei ole olemassa segmenttejä joissa geometriaa ei ole. Silloin on vain alignment jolla ei ole segmenttejäkään. Eli kun tätä haetaan per-segment, on varmaan turha kutsua koko funktiota jos segmenttiä ei ole (mitään muutakaan ei voi silloin löytyä), esim suodattamalla where-ehdossa pois rivit joissa on pelkkä alignment: `and segment_index = null` tai käyttämällä inner joinia